ui/tests/integration/components/confirm-action-test.js

```diff
@@ -0,0 +1,53 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render, click } from '@ember/test-helpers';
+import hbs from 'htmlbars-inline-precompile';
+import sinon from 'sinon';
+
+module('Integration | Component | confirm-action', function (hooks) {
+  setupRenderingTest(hooks);
+
+  test('it renders and on click shows the correct icon', async function (assert) {
+    let confirmAction = sinon.spy();
+    this.set('onConfirm', confirmAction);
+    await render(hbs`
+      <ConfirmAction
+        @onConfirmAction={{onConfirm}}
+        @buttonClasses="toolbar-link"
+      >
+        DELETE
+        </ConfirmAction>
+      `);
+    assert.dom('[data-test-icon="chevron-down"]').exists('Icon is pointing down');
+    await click('[data-test-confirm-action-trigger="true"]');
+    assert.dom('[data-test-icon="chevron-up"]').exists('Icon is now pointing up');
+    assert.dom('[data-test-confirm-action-title]').hasText('Delete this?');
+  });
+
+  test('it closes the confirmation modal on successful delete', async function (assert) {
+    let confirmAction = sinon.spy();
+    this.set('onConfirm', confirmAction);
+    await render(hbs`
+      <ConfirmAction
+        @onConfirmAction={{onConfirm}}
+        @buttonClasses="toolbar-link"
+      >
+        DELETE
+        </ConfirmAction>
+      `);
+    await click('[data-test-confirm-action-trigger="true"]');
+    await click('[data-test-confirm-cancel-button="true"]');
+    // assert that after canceling the icon button is pointing down.
+    assert.dom('[data-test-icon="chevron-down"]').exists('Icon is pointing down after clicking cancel');
+    // open the modal again to test the Cancel action
```

Comment:
we're testing the `confirm` action (Delete here) right?

Reply:
Yes, gosh, coffee needed because these two words apparently are the same to me this morning. Thank you. Amended the comment.